client/ayon_maya/plugins/load/load_layout.py

```diff
@@ -0,0 +1,231 @@
+                        all_loaders, repre_id)
+
+                    loader = None
+
+                    if repr_format:
+                        loader = self._get_loader(loaders, product_type)
+
+                    if not loader:
+                        self.log.error(
+                            f"No valid loader found for {repre_id}")
+                        continue
+
+                    options = {
+                        # "asset_dir": asset_dir
+                    }
+                    load_container(
+                        loader,
+                        repre_id,
+                        namespace=instance_name,
+                        options=options
+                    )
+                instances = [
+                    item for item in data
+                    if ((item.get('version') and
+                        item.get('version') == element.get('version')))]
+
+                for instance in instances:
+                    transform = instance.get('transform')
+                    instance_name = instance.get('asset_name')
+                    extension = instance_name.split("_")[-1]
+                    instance_name = instance_name.replace(f"_{extension}", "")
+                    self.set_transformation(instance_name, transform)
+
+    def set_transformation(self, instance_name, transform):
+        asset = self.get_asset(instance_name)
+        translation = [
+            transform["translation"]["x"],
+            transform["translation"]["z"],
+            transform["translation"]["y"]
+            ]
+
+        rotation = [
+            math.degrees(transform["rotation"]["x"]),
+            180 - math.degrees(transform["rotation"]["z"]),
+            math.degrees(transform["rotation"]["y"]),
+        ]
+        scale = [
+            -transform["scale"]["x"],
+            transform["scale"]["z"],
+            transform["scale"]["y"]
+        ]
+
+        cmds.xform(
+            asset,
+            translation=translation,
+            rotation=rotation,
+            scale=scale
+        )
+
+    def load(self, context, name, namespace, options):
+
+        path = self.filepath_from_context(context)
+
+        self.log.info(">>> loading json [ {} ]".format(path))
+        self._process(path, options)
+        folder_name = context["folder"]["name"]
+        namespace = namespace or unique_namespace(
+            folder_name + "_",
+            prefix="_" if folder_name[0].isdigit() else "",
+            suffix="_",
+        )
+        # TODO: implement the function to load all the assets and set transforms
+        return containerise(
+            name=name,
+            namespace=namespace,
+            nodes=[],
+            context=context,
+            loader=self.__class__.__name__)
+
+    def update(self, container, context):
+        repre_entity = context["representation"]
+        path = get_representation_path(repre_entity)
+        self._process(path, options=None)
```

Comment:
Pretty sure updates won't work here - this will just load everything AGAIN on update?

Reply:
it won't load everything again. it would only update on the transform. There is a check on whether the loaded container exists before loading the assets.

Reply:
But what if objects were removed from layout? Or if new ones were added?

Also - I think the code may become more readable if somehow you can have a `process_element` or whatever 'element' is so you can reduce the nesting a bit. Like along the lines of:
```python
for element in data:
    self.process_element(element)
```

Reply:
> But what if objects were removed from layout? Or if new ones were added?

this is something we should consider for not only the loader here but also the one lasted in unreal